src/couch_server.erl

```diff
@@ -145,6 +150,8 @@ maybe_add_sys_db_callbacks(DbName, Options) ->
 path_ends_with(Path, Suffix) ->
     Suffix == lists:last(binary:split(mem3:dbname(Path), <<"/">>, [global])).
 
+check_dbname(#server{dbname_validator={M, F, A}}, DbName) ->
+    M:F(DbName, A);
 check_dbname(#server{dbname_regexp=RegExp}, DbName) ->
```

Comment:
Why do we have to maintain two different ways to apply dbname validation logic (MFA vs regex)? May be drop dbname_regexp and replace it with the default MFA which points to the original check_dbname? Would be a bit more consistent.